elastic/kibana
# [Fleet] Make stream id unique in agent policy

## Description 

Resolve #81617 https://github.com/elastic/beats/issues/22054

This PR make stream id unique, this allow to have multiple time the same integrations in an agent policy.

What changed?
* the stream id is not generated client side anymore
* when we persist a package policy (create|bulkCreate|update) we generated ids for streams
* the stream id is suffixed the package policy id.
* For input validate client side we do not use id anymore but `dataset`

I confirmed with @michalpristas there is not limit on the id size.

## Example config

### After

<img width="738" alt="Screen Shot 2020-11-03 at 4 56 56 PM" src="https://user-images.githubusercontent.com/1336873/98046961-26cd8f00-1df9-11eb-972c-36856a94efde.png">


### Before

<img width="687" alt="Screen Shot 2020-11-03 at 5 21 19 PM" src="https://user-images.githubusercontent.com/1336873/98046977-2af9ac80-1df9-11eb-9525-94d5d541d7e5.png">


## How to test this?

* Create a config 
* add multiple time the same integration (custom log is easy to read and configure)
* Copy a policy should work too
